### PR TITLE
DEVPROD-7742 Fix lint warnings and errors

### PR DESCRIPTION
**Pull request by ZackarySantana**
DEVPROD-7742

### Description
This fixes the various lint warnings and errors we have throughout our project. I'm going to open a 2nd PR to try to fix up a task to fail PR's on this lint warnings and errors, this is to keep the work separated.

### Testing
Existing unit tests. Nothing stood out to me to manually test, it's mostly in test files. The ones outside of test files is just removing an unused variable, preventing a nil dereference, and other things that manual checks aren't needed for.

### CODE REVIEW

File: rest/route/patch_test.go [hunk 1]
```diff
@@ -20,7 +20,6 @@ import (
 	"github.com/evergreen-ci/evergreen/model/task"
 	"github.com/evergreen-ci/evergreen/model/user"
 	"github.com/evergreen-ci/evergreen/rest/data"
-	"github.com/evergreen-ci/evergreen/rest/model"
 	restmodel "github.com/evergreen-ci/evergreen/rest/model"
```
Comment (ablack12) [inline rest/route/patch_test.go:23]: nit: can we change this to restModel to be consistent with serviceModel?

**Review by ablack12 — APPROVED** [review]
LGTM -- ideally we could fix up the existing lint tasks to be stricter rather than adding new tasks.

**Comment by ZackarySantana** [comment]
Hm we could just enable the lint rules that I applied to 'fix' the codebase, I wasn't sure if we wanted it to be as strict. If it sounds good to you, I'll include that in a separate PR (in case we need to revert, it doesn't revert these lint fixes)?

**Comment by ablack12** [comment]
Yeah separate PR sounds good to me; I think we may as well just be strict.

**Pull request merged**